tests/spcs/test_services.py

```diff
@@ -699,6 +699,401 @@ def test_logs_streaming_disabled(mock_is_disabled, runner):
+        "                        )\n"
+        "                        and record_type = 'LOG'\n"
+        "                        and scope['name'] = 'snow.spcs.platform'\n"
+        "                        order by timestamp desc\n"
+        "                        limit 10\n"
+        "                    )\n"
+        "                    order by timestamp asc\n"
+        "                    \n"
+        "                "
+    )
+
+    assert (
+        actual_query == expected_query
+    ), f"Generated query does not match expected query.\n\nActual:\n{actual_query}\n\nExpected:\n{expected_query}"
+
+
+@patch(
+    "snowflake.cli.api.feature_flags.FeatureFlag.ENABLE_SPCS_SERVICE_EVENTS.is_disabled"
+)
+def test_events_first_last_incompatibility(mock_is_disabled, runner):
+    mock_is_disabled.return_value = False
+    result = runner.invoke(
+        [
+            "spcs",
+            "service",
+            "events",
+            "LOG_EVENT",
+            "--container-name",
+            "log-printer",
+            "--instance-id",
+            "0",
+            "--first",
+            "10",
+            "--last",
+            "5",
+            "--warehouse",
+            "XSMALL",
+            "--role",
+            "sysadmin",
+        ]
+    )
+
+    assert result.exit_code != 0, result.output
+
+    expected_error = "Parameters '--first' and '--last' are incompatible"
+    assert expected_error in result.output
+
+
+@patch(
+    "snowflake.cli.api.feature_flags.FeatureFlag.ENABLE_SPCS_SERVICE_METRICS.is_disabled"
+)
+def test_service_metrics_disabled(mock_is_disabled, runner):
+    mock_is_disabled.return_value = True
+    result = runner.invoke(
+        [
+            "spcs",
+            "service",
+            "metrics",
+            "LOG_EVENT",
+            "--container-name",
+            "log-printer",
+            "--instance-id",
+            "0",
+            "--since",
+            "1 minute",
+        ]
+    )
+    assert (
+        result.exit_code != 0
+    ), "Expected a non-zero exit code due to feature flag being disabled"
+
+    expected_output = (
+        "+- Error ----------------------------------------------------------------------+\n"
+        "| Service metrics collection from SPCS event table is disabled. To enable it,  |\n"
+        "| add 'ENABLE_SPCS_SERVICE_METRICS = true' to '[cli.features]' section of your |\n"
+        "| configuration file.                                                          |\n"
+        "+------------------------------------------------------------------------------+\n"
+    )
+    assert (
+        result.output == expected_output
+    ), f"Expected formatted output not found: {result.output}"
+
+
+@patch(
+    "snowflake.cli.api.feature_flags.FeatureFlag.ENABLE_SPCS_SERVICE_METRICS.is_disabled"
+)
+@patch("snowflake.cli._plugins.spcs.services.manager.ServiceManager.execute_query")
+def test_latest_metrics(mock_execute_query, mock_is_disabled, runner):
```

Comment:
Can we check what rows were printed as well (to check whether formatting / show all columns flag works as expected). `snapshot` fixture and `pytest.mark.parametrize` might be useful - you can check out `test_help_messages` for usage example :)